src/Interpreters/executeQuery.cpp

```diff
@@ -289,6 +293,29 @@ addStatusInfoToQueryLogElement(QueryLogElement & element, const QueryStatusInfo
     element.async_read_counters = context_ptr->getAsyncReadCounters();
 }
 
+static void checkCPULoad(const ContextPtr context)
+{
+    /// We need to get last_value first to avoid the effect of a possible data race, when current loaded value will be less than the previous one
+    auto last_wait_time = ProfileEvents::global_counters.getLastValue(ProfileEvents::OSCPUWaitMicroseconds);
+    auto wait_time = ProfileEvents::global_counters[ProfileEvents::OSCPUWaitMicroseconds].load(std::memory_order_relaxed) - last_wait_time;
+    auto last_busy_time = ProfileEvents::global_counters.getLastValue(ProfileEvents::OSCPUVirtualTimeMicroseconds);
+    auto busy_time = ProfileEvents::global_counters[ProfileEvents::OSCPUVirtualTimeMicroseconds].load(std::memory_order_relaxed) - last_busy_time;
```

Comment:
Also, it may get inconsistent values since it's not atomic

Reply:
Yeah, I know, but isn't it acceptable in this case? I don't want to introduce lock here to get consistent values. The worst effect is we underestimate the CPU load (since we get the wait time first)

Reply:
> The worst effect is we underestimate the CPU load (since we get the wait time first)

Hm, yes, it's okay, but it requires a comment